src/weather_type.cpp

```diff
@@ -110,6 +111,7 @@ void weather_type::load( const JsonObject &jo, const std::string & )
     mandatory( jo, was_loaded, "ranged_penalty", ranged_penalty );
     mandatory( jo, was_loaded, "sight_penalty", sight_penalty );
     mandatory( jo, was_loaded, "light_modifier", light_modifier );
+    mandatory( jo, was_loaded, "load_order", load_order );
```

Comment:
Is there a reason this is exposed to JSON instead of something like
```suggestion
    static int current_load_order = -1;
    load_order = current_load_order;
    ++current_load_order;
```

Reply:
Yes, its the whole point of the PR in some ways.  The order a weather type is loaded in is arbitrary especially when dealing with mods.  But its not a meaningless field, it matters in the logic in terms of what weather it is.  If the conditions for two or more weather types are true then the one with the higher load order is what happens.

Reply:
Ah, my bad. Perhaps this field would be better named `priority`.